open-apparel-registry/open-apparel-registry
# Adjust map style and behaviors

## Overview

- add custom markers and styles for clusters
- add and use custom marker for static image API
- use custom markers for facilities markers
- zoom map to marker on selecting it from the list or visiting it
directly

Connects #302 
Connects #244 
Connects #128 

## Demo

![markers](https://user-images.githubusercontent.com/4165523/54388835-0e818d00-4675-11e9-8e9a-acdfc4f259c2.gif)
![markers-2](https://user-images.githubusercontent.com/4165523/54388836-0f1a2380-4675-11e9-8f09-df612310d1b1.gif)


## Notes

As discussed in Slack we opted to keep the static endpoint and send a marker URL location, which will be the images dir from staging and production. In development, however, this will not work. To test this locally:

- run a static server from `src/app/public` in this repository
- ngrok http <whatever port>
- replace `icon:${window.location.origin}` with `icon:https://whatever.ngrok.io` in the FacilityDetailsStaticMap component, leaving `/images/static-marker.png.....` in place.

## Testing Instructions

- serve this branch
- set up ngrok as described in the notes field above
- visit the map and interact with it. verify that you see the new markers for clusters, the JS API, and the static API, and that interacting with the facilities list through the filter sidebar zooms the map to a location, but clicking on a marker on the map main does not zoom
